internal/security/bootstrapper/config/types.go

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package config
+
+// BootStrapperInfo defines the first stage gate info
+// It is the first stage gate of security bootstrapping
+type BootStrapperInfo struct {
+	Host      string
+	StartPort int
+}
+
+// ReadyInfo defines the ready stage gate info
+// It is the last stage gate of security bootstrapping for
+// Kong, and all other Edgex core services
+type ReadyInfo struct {
+	ToRunPort int
+}
+
+// TokensInfo defines the tokens ready stage gate info
+// for the secretstore (aka. vault-worker)
+type TokensInfo struct {
+	ReadyPort int
+}
+
+// VaultWorkerInfo defines the fields related to
+// stage gating the secretstore, vault-worker
+type VaultWorkerInfo struct {
+	Host   string
+	Tokens TokensInfo
+}
+
+// RedisInfo defines the fields related to
+// stage gating the redis db
+type RedisInfo struct {
```

Comment:
Similar here, IMO this should be referred to as Database, rather than pinning it to Redis.

Reply:
I don't want the whole Database structure though: 
```toml
[Databases]
  [Databases.Primary]
  Host = 'localhost'
  Name = 'metadata'
  Port = 6379
  Timeout = 5000
  Type = 'redisdb'
```
plus, in the `security-bootstrapper` scenario, there is different `ReadyPort` in addition to the original structure. So I prefer to shun away from that definition of struct.

Reply:
No, I didn't mean that, just the name used here. `RedisInfo ` => `DatabaseInfo`

Reply:
But you do make a good point, we already have `Host` and `Port` in `DATABASES_.PRIMARY` so all we need here is `ReadyPort`

Reply:
but how do I use that in the go code if i don't have that Database structure?